app/grandchallenge/challenges/forms.py

```diff
@@ -534,3 +535,26 @@ def __init__(self, *args, **kwargs):
             self.fields["status"].widget.attrs["disabled"] = True
         self.helper = FormHelper(self)
         self.helper.layout.append(Submit("save", "Save"))
+
+    def clean_status(self):
```

Comment:
The responsibility of a `clean_` method is only to clean the forms field, too much is happening here. Please think carefully about what a method should do, and where things should be put. You will likely need to split up the functionality here.